### DISCUSSION TITLE
How to have computed properties reevaluated when their underlying parameters change?

### DISCUSSION THREAD

**Discussion by kslstn** [alpinejs/alpine#3212] [1. Help] [answered]
Let's say I have a property `positionX`, which is defined something like:
```
get positionX(){ return get document.getElementbyId('someID').offsetLeft; }
```
Now when the DOM changes (when the element with `someID` is moved, or changes position because the layout changes with a viewport size change), I need `positionX` to be updated. Is there anyway to give Alpine a nudge to do that? I know the example is dumb, because I could define `someID`s position via properties in Alpine. I've got a more realistic scenario in [this Codepen](https://codepen.io/kslstn/pen/QWrPqoO).

**Comment by SimoTod** [comment, marked as answer]
Alpine reacts to changes to property defined in x-data. A change to a dom element position won't trigger the reactivity. However, you can use transitionIsInProgress to make sure the value is recalculated when the animation starts. You just need to access it in your computed prop.
```js
 get boxWidth(){
     this.transitionIsInProgress;
			return window.getComputedStyle(this.$refs[this.currentBox]).getPropertyValue('width');
		},
```
